events.go

```diff
@@ -214,6 +216,41 @@ var fakeCancelReason = []byte("internal: cancel")
 var fakeTimeoutReason = []byte("internal: call state timeout")
 var fake2xxReason = []byte("internal: implied OK")
 
+func (d *EventData) EncryptIP(key [16]byte) error {
+	var err error
+	if d.Src.To4() != nil && d.Dst.To4() != nil {
+		err = d.EncryptIPv4(key)
+	} else if d.Src.To16() != nil && d.Dst.To16() != nil {
+		err = d.EncryptIPv6(key)
+	} else {
+		err = errors.New("EncryptIP: broken IP address")
+	}
+	return err
+}
+
```

Comment:
while it is a good idea to have an option for an internal event to encrypt itself, I think it would be better to add the encryption only on the published events (this way if I look on the web interface 127.0.0.1:8080, I will see the real IPs).
This would mean adding the encryption to sipcmbeat. For the IPs the right place would be:
https://github.com/intuitivelabs/sipcmbeat/blob/master/beater/sipcmbeat.go#L314
and
https://github.com/intuitivelabs/sipcmbeat/blob/master/beater/sipcmbeat.go#L316 .

We could also leave it this way and add ed.EncryptIP(...) at
https://github.com/intuitivelabs/sipcmbeat/blob/master/beater/sipcmbeat.go#L250 , since there ed is a copy, but wouldn't it be more clear to encrypt each added field in place in sipcmbeat, rather then relying on some per event Encrypt() calls?

Reply:
I was actually planning to encrypt the IPs in `Sipcmbeat.consumeEv()`, like this:

```
func (bt *Sipcmbeat) consumeEv() {
    defer bt.wg.Done()
waitsig:
    for {
        select {
        case <-bt.done:
            return
        case <-bt.newEv:
            stats.Inc(cntEvSigs)
            last := bt.evRing.LastIdx()
            stats.Max(cntEvMaxQ, counters.Val(last-bt.evIdx))
            for bt.evIdx != last {
                ev, nxtIdx, err := bt.evRing.Get(bt.evIdx)
                if ev != nil {
                    if err := ev.EncryptIP(bt.anonymKey); err != nil {
                        fmt.Fprintf(os.Stderr, "could not anonymize IP address; error: %s\n", err)
                    }
                    bt.publishEv(ev)
                    bt.evRing.Put(bt.evIdx)
...

but it looks like I do not understand the `calltr.EventData` life cycle...
```

Reply:
There is one event ring that can have several "views" on it.  An event is never deleted from the ring. Each ring "viewer" keeps the index of the last event seen and request newer events each time.
This allows to have sipcmbeat and the sipcallmon web interface use the event ring in parallel, without influencing each other (and theoretically allows having multiple independent events publisher).
You should not modify and event in the ring (unless you want every other possible ring viewer to see the change), so that's why you should not encrypt the event got from the ring, but either directly the produced ES fields or a copy of the event.